src/pandapipes/pf/pipeflow_setup.py

```diff
@@ -645,28 +608,20 @@ def check_connectivity(net, branch_pit, node_pit, mode="hydraulics"):
     :rtype: tuple(np.array)
     """
     if mode == "hydraulics":
-        active_branch_lookup = branch_pit[:, ACTIVE_BR].astype(np.bool_)
-        active_node_lookup = node_pit[:, ACTIVE_ND].astype(np.bool_)
-        slacks = np.where((node_pit[:, NODE_TYPE] == P) & active_node_lookup)[0]
+        slacks = np.where((node_pit[:, NODE_TYPE] == P) & nodes_connected)[0]
     else:
-        if get_net_option(net, "transient"):
-            active_branch_lookup = get_lookup(net, "branch", "active_hydraulics")
-        else:
-            active_branch_lookup = branches_connected_flow(branch_pit) \
-                                   & get_lookup(net, "branch", "active_hydraulics")
-        active_node_lookup = node_pit[:, ACTIVE_ND].astype(np.bool_) \
-                             & get_lookup(net, "node", "active_hydraulics")
-        slacks = np.where((node_pit[:, NODE_TYPE_T] == T) & active_node_lookup)[0]
+        slacks = np.where(((node_pit[:, NODE_TYPE_T] == T) | (node_pit[:, NODE_TYPE_T] == GE)) & nodes_connected)[0]
 
-    return perform_connectivity_search(net, node_pit, branch_pit, slacks, active_node_lookup,
-                                       active_branch_lookup, mode=mode)
+    return perform_connectivity_search(net, node_pit, branch_pit, slacks,
+                                       nodes_connected, branches_connected, mode=mode)
 
 
 def perform_connectivity_search(net, node_pit, branch_pit, slack_nodes, active_node_lookup, active_branch_lookup,
                                 mode="hydraulics"):
     connect = branch_pit[:, FLOW_RETURN_CONNECT].astype(bool)
-    circ = branch_pit[:, BRANCH_TYPE] == CIRC
-    if np.any(circ) and mode == 'hydraulics':
+    to_nodes = branch_pit[:, TO_NODE]
+    gen = node_pit[to_nodes.astype(int), NODE_TYPE_T] == GE
+    if np.any(gen) and mode == 'hydraulics':
```

Comment:
Why do we need this gen check here? Is there no other constellation where FLOW_RETURN_CONNECT needs to be considered? What about incomplete networks?